internal/utils/helm/package.go

```diff
@@ -0,0 +1,493 @@
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/kubectl/pkg/cmd/util"
+
+	"github.com/envoyproxy/gateway/internal/cmd/options"
+)
+
+const (
+	helmOperateTimeout = time.Second * 300
+	egChartVersion     = "v1.0.0"
+	egReleaseName      = "envoy-gateway"
+	egReleaseNamespace = "envoy-gateway-system"
+)
+
+type PackageOptions struct {
+	DryRun           bool
+	SkipCRD          bool
+	Wait             bool
+	Version          string
+	Timeout          time.Duration
+	ReleaseName      string
+	ReleaseNamespace string
+	OnlyCRD          bool
+	WithCRD          bool
+}
+
+type PackageTool struct {
+	chartName string
+
+	// Helm dependency objects
+	envSettings     *cli.EnvSettings
+	actionConfig    *action.Configuration
+	actionInstall   *action.Install
+	actionUninstall *action.Uninstall
+	valuesOpts      *values.Options
+
+	logger Printer
+}
+
+func NewPackageTool() *PackageTool {
+	return &PackageTool{
+		envSettings:  cli.New(),
+		actionConfig: &action.Configuration{},
+		chartName:    "oci://docker.io/envoyproxy/gateway-helm",
+		valuesOpts:   &values.Options{},
+	}
+}
+
+// Setup Configuration required to initialize helm action.
+func (pt *PackageTool) Setup() error {
+
+	// Since envoy-gateway uses docker's oci to store charts,
+	// we need to create a registry client to make sure we can retrieve envoy-gateway chart
+	registryCli, err := registry.NewClient()
+	if err != nil {
+		return err
+	}
+	pt.actionConfig = &action.Configuration{
+		RegistryClient: registryCli,
+	}
+
+	kubectlFactory := options.DefaultConfigFlags
+	ns, _, err := kubectlFactory.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	if err = pt.actionConfig.Init(
+		kubectlFactory,
+		ns,
+		os.Getenv("HELM_DRIVER"),
+		pt.logger.Printf,
+	); err != nil {
+		return err
+	}
+
+	// Build the relevant helm command action
+	pt.actionInstall = action.NewInstall(pt.actionConfig)
+	pt.actionUninstall = action.NewUninstall(pt.actionConfig)
+
+	return nil
+}
+
+// SetInstallEnvSettings set the installation flags we are interested in
+func (pt *PackageTool) SetInstallEnvSettings(installCmd *cobra.Command, opts *PackageOptions) {
+
+	// add helm flags
+	// we use a temporary flag to be set by helm env flags,
+	// from which we can retrieve the flags we are interested
+	var tmpFlags pflag.FlagSet
+	pt.envSettings.AddFlags(&tmpFlags)
+	tmpFlags.VisitAll(func(flag *pflag.Flag) {
+		// TODO: Add more flags as needed?
+		switch flag.Name {
+		case "registry-config", "repository-config", "repository-cache":
+			installCmd.Flags().AddFlag(flag)
+		default:
+		}
+	})
+
+	installCmd.Flags().DurationVar(&opts.Timeout, "timeout", helmOperateTimeout, "time to wait for any individual Kubernetes operation")
+	installCmd.Flags().StringVar(&opts.Version, "version", egChartVersion, "specify a version constraint for the envoy gateway version to use")
+	installCmd.Flags().StringVar(&opts.ReleaseName, "release-name", egReleaseName, "name of the envoy-gateway release to install")
```

Comment:
maybe just `name` ?

Reply:
> maybe just `name` ?

I have thought about this problem, but I think it is more suitable to use `release-name`.

+ it's self-explanatory
+ release-name does not seem to be easily ambiguous, at least when compared with name.

Reply:
we ideally like typing less when its comes to CLI commands, and the flag description does mention that this is related to release.
Lets get this in, and get feedback, since its experimental, we can always improve this later